src/Features/Core/Portable/Completion/CompletionHelper.cs

```diff
@@ -229,15 +234,17 @@ private int CompareMatches(PatternMatch match1, PatternMatch match2, CompletionI
                 return diff;
             }
 
-            // Now, after comparing matches, check if an item wants to be preselected.  If so,
-            // we prefer that.  i.e. say the user has typed 'f' and we have the items 'foo' 
-            // and 'False' (with the latter being 'Preselected').  Both will be a prefix match.
-            // And because we are ignoring case, neither will be seen as better.  Now, because
-            // 'False' is preselected we pick it even though 'foo' matches 'f' case sensitively.
-            diff = item2.Rules.MatchPriority - item1.Rules.MatchPriority;
-            if (diff != 0)
+            // If they both seemed just as good, but they differ on preselection, then
```

Comment:
Removed the redundant check from CompareItems--it calls CompareMatches which already does this check.